feat: Enabled PWA support

meghadeep-com opened the pull request:
Just added the manifest.json file, the PWA icons, and the manifest link in the index.html file.

This enables PWA support for Gecko (tested), Chromium (tested), and WebKit (untested) based browsers.

The PWA icons need to be updated before merging this request, it's pretty much hand-drawn with a mouse from the original favicon.ico. If you provide me with at least a 512x512 image of the logo, I can regenerate the PWA icons.

rabilrbl commented:
I'm genuinely impressed that this was hand-drawn using a mouse!
The existing logo was just a generic one I found online.
Feel free to change the logo, just make sure it doesn’t infringe on Jio’s copyright.

meghadeep-com commented on web/static/manifest.json:
```diff
@@ -0,0 +1,465 @@
+  "theme_color": "#000000",
+  "background_color": "#000000",
+  "short_name": "JioTV Go",
+  "id": "?q=high",
```
The ID needs to be different according to personal implementations, since different users will necessarily have different public hostnames. The ID can be "/" instead, but not "jiotv-go".

meghadeep-com commented on web/static/manifest.json:
```diff
@@ -0,0 +1,465 @@
+  "background_color": "#000000",
+  "short_name": "JioTV Go",
+  "id": "?q=high",
+  "start_url": "/?q=high",
```
Fair.

meghadeep-com commented:
@rabilrbl Fortunately there's a service called PWAbuilder that lets me generate the whole icon set needed for PWA support from a single image. Updated the icon set with a royalty free (no existing license, not even CC-BY-SA) images. And updated the start URL and the app ID to be more generic (no defaulting to high quality).

rabilrbl commented:
Will do a detailed review when I get time.

rabilrbl commented:
@meghadeep-com Sorry to tell this now, but let's remove the JioTV Go text from the logo. We might change the project name some day, I don't want this to be a dependency.

rabilrbl commented on web/static/manifest.json:
```diff
@@ -0,0 +1,465 @@
+  "id": "/",
+  "start_url": "/",
+  "icons": [
+    {
```
@meghadeep-com The concern isn't maintenance here. We ship release binaries in a single packed file that will also include all web assets and having more logos just for PWA is a issue here.
Let's only keep the bare minimum required.

rabilrbl commented on web/static/manifest.json:
```diff
@@ -0,0 +1,465 @@
+      "sizes": "1024x1024"
+    }
+  ],
+  "display": "standalone",
```
How will this work with Mobile users who will start the app in Portrait?

rabilrbl commented on web/static/manifest.json:
```diff
@@ -0,0 +1,579 @@
+{
+  "name": "JioTV Go",
+  "theme_color": "#000000",
+  "background_color": "#000000",
```
We need to to give more thought here.

How will it work for light and dark mode in our case?

rabilrbl commented on web/static/manifest.json:
```diff
@@ -0,0 +1,579 @@
+  "theme_color": "#000000",
+  "background_color": "#000000",
+  "short_name": "JioTV Go",
+  "id": "/",
```
Shoudn't this be a unique identifier? Why a path?

meghadeep-com replied:
The manifest.json file follows the browser cache directives (since I have not added any overrides). So, any updates to the JioTV Go server will be picked up the next time the PWA reloads, or when the browser cache expires.

rabilrbl reviewed (changes requested):
Largely looks good, minor nitpicks.

Are there any other efficient formats supported besides PNG, such as WebP?

rabilrbl commented on web/static/manifest.json:
```diff
@@ -0,0 +1,37 @@
+      "type": "image/webp"
+    },
+    {
+      "src": "/static/icons/icon-medium.ico",
+      "sizes": "72x72 96x96 128x128 256x256",
+      "type": "image/vnd.microsoft.icon"
+    },
+    {
```
@meghadeep-com Let's stick to webp perhaps?

rabilrbl replied:
Yup, I'm fine with that.

rabilrbl commented on web/static/manifest.json:
```diff
@@ -0,0 +1,32 @@
+    },
+    {
+      "src": "/static/icons/icon-high.svg",
+      "sizes": "257x257",
```
I think this is better
```suggestion
      "sizes": "any",
```

meghadeep-com replied:
Not a typo. This will result in any icon larger than 256x256 to be served by the SVG file instead, which supports scaling.

rabilrbl commented on web/static/manifest.json:
```diff
@@ -0,0 +1,32 @@
+    },
+    {
+      "src": "/static/icons/icon-medium.png",
+      "sizes": "72x72 96x96 128x128 256x256",
+      "type": "image/png"
+    },
+    {
```
Let me see what we can do here.

meghadeep-com replied:
Eh, Copilot is being a bit too nitpicky, Mozilla is fine with this: https://developer.mozilla.org/en-US/docs/Web/Progressive_web_apps/How_to/Define_app_icons#create_the_necessary_icon_sizes

rabilrbl commented:
Oh wait, only SVG is scalable
